src/sagas/wallet.js

```diff
@@ -287,12 +291,17 @@ export function* loadTokens() {
   // We don't need to wait for the metadatas response, so just fork it
   yield fork(fetchTokensMetadata, registeredTokens);
 
-  // Since we already know here what tokens are registered, we can dispatch actions
-  // to asynchronously load the balances of each one. The `put` effect will just dispatch
-  // and continue, loading the tokens asynchronously
-  for (const token of registeredTokens) {
+  // Dispatch actions to asynchronously load the balances of each token the wallet has
+  // ever interacted with. The `put` effect will just dispatch and continue, loading
+  // the tokens asynchronously.
+  //
+  // Note: We need to download the balance of all the tokens from the wallet so we can
+  // hide zero-balance tokens
+  for (const token of allTokens) {
     yield put(tokenFetchBalanceRequested(token));
```

Comment:
After we finish fetching the balance, do we fetch the history automatically? If that's the case, then we could fetch the history only from the registered tokens.

Reply:
We only fetch the history when the user clicks the token to see more details, we don't pre-fetch token history